### DISCUSSION TITLE
divinum-replay and divinum-get

### DISCUSSION THREAD

**Discussion by ajmalton** [DivinumOfficium/divinum-officium#2577] [Developers corner]
I'm going to add cookie support to the regression testing tools in admin/.  (I'm not looking at those in regress/ at all.)

I'm going to make -get accept a cookie jar as input, sending it to the server, and recording both sent and received cookies on the output file.

I'm going to make -replay resend the previously _sent_ cookies, and verify that re-received cookies are "similar enough" -- probably that the names match and the values match -- to the recorded ones.

This results in a slight ambiguity with the previous -replay, because saved tests with no cookie might be old style (and didn't store any cookie) or new style (and didn't get a cookie). I'll add --filter=-cookies to suppress the comparison, but that will not be the default.

I have to do this now because some of my changes have affected cookie handling slightly -- and my changes are supposed to be net-zero end-to-end.

Does this plan bother anyone?

**Comment by gregordick** [comment]
Sounds good to me.

**Comment by ajmalton** [comment]
This was released as https://github.com/DivinumOfficium/divinum-officium/commit/b89792809de170589c1e7803e721a71bc1485089
